### PR TITLE
Migrate Grayscale to use BaseImageAugmentationLayer

### LINKED ISSUE #191: Migrate Grayscale to use BaseImageAugmentationLayer
https://github.com/keras-team/keras-cv/blob/master/keras_cv/layers/preprocessing/grayscale.py

### PR DESCRIPTION
**Pull request by quantumalaviya**
closes #191

### CODE REVIEW

File: keras_cv/layers/preprocessing/grayscale.py [hunk 1]
```diff
@@ -13,19 +13,17 @@
...
 
 
-class Grayscale(layers.Layer):
+class Grayscale(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
     """Grayscale is a preprocessing layer that transforms RGB images to Grayscale images.
     Input images should have values in the range of [0, 255].
 
     Input shape:
-        3D (unbatched) or 4D (batched) tensor with shape:
+        4D (batched) tensor with shape:
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/grayscale.py:23]: revert - can be either unbatched or batched

**Review by LukeWood — CHANGES_REQUESTED** [review]

**Comment by LukeWood** [comment]
Thanks for the PR @quantumalaviya - only one comment.

File: keras_cv/layers/preprocessing/grayscale.py (hunk 1 above)
Comment (quantumalaviya) [reply keras_cv/layers/preprocessing/grayscale.py]: done!

**Comment by quantumalaviya** [comment]
My pleasure! It's a great learning experience.

**Pull request merged**